adding support for pulling data off of devices.

pokey909 opened the pull request:
Pulled data will be stored on the jenkins master as artifacts.

*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

pokey909 committed:
adding support for pulling data off of devices. Fix comma separated filename parsing.
Pulled data will be stored on the jenkins master as artifacts.

nikhil-dabhade committed:
Merge pull request #18 from awslabs/master

Update base with fixes

nikhil-dabhade committed:
[maven-release-plugin] prepare release aws-device-farm-1.28

nikhil-dabhade committed:
[maven-release-plugin] prepare for next development iteration

nikhil-dabhade committed:
[maven-release-plugin] rollback the release of aws-device-farm-1.28

nikhil-dabhade committed:
[maven-release-plugin] prepare release aws-device-farm-1.28

nikhil-dabhade committed:
Update version to 1.29-SNAPSHOT

nikhil-dabhade committed:
[maven-release-plugin] prepare release aws-device-farm-1.29

nikhil-dabhade committed:
[maven-release-plugin] prepare for next development iteration

nikhil-dabhade commented on pom.xml:
```diff
@@ -8,7 +8,7 @@
   <groupId>org.jenkins-ci.plugins</groupId>
   <artifactId>aws-device-farm</artifactId>
-  <version>1.27-SNAPSHOT</version>
+  <version>1.28-SNAPSHOT</version>
```
This file should not be manually edited as it is auto-updated with Jenkins release.

awsshrey commented:
@pokey909 This pull request was tested at our end and is working fine. However, requesting you to please add unit test for the change before we can merge to mainline.

nikhil-dabhade committed:
Add support for chained IAM roles

- Update the timeout to 1 hr for chained role
- Use AWSDeviceFarmClientBuilder instead of AWSDeviceFarmClient
- Thanks to @ramapalani for this contribution

nikhil-dabhade committed:
Add logging for invalida credentials

nikhil-dabhade committed:
Merge pull request #21 from nikhil-dabhade/master

Add support for chained IAM roles

nikhil-dabhade committed:
[maven-release-plugin] prepare release aws-device-farm-1.30

nikhil-dabhade committed:
[maven-release-plugin] prepare for next development iteration

pokey909 commented:
@awsshrey Good to hear. This is also running in production for soundcloud for >2years now.
However, I don't really know how to test this properly within the whole jenkins testing framework.
Could you help adding a short unit test?
